src/app/core/services/users.service.ts

```diff
@@ -65,48 +70,136 @@ export class UsersService {
       albums: [],
     };
 
-    for (const game of userData.document.collection.games) {
-      const gameData = await game.ref.get();
-      if (game.in_collection) {
-        collection.games.push({
-          general: {
-            ...gameData.data(),
-            id: game.ref.id,
-          } as GameWithId,
-          user: game,
-        });
-      }
-    }
+    await Promise.all(
```

Comment:
I'm running into a performance problem here. To get the names, ids and additional data for each collection entry I have to retrieve that data for every single entry. Which means a large collection with e.g. 200 entries will perform 200 requests. Other document DBs like mongoDB have the functionality to populate an array of document references at once, which streamlines the process substantially. Firebase does not seem to offer this, so to improve performance here I'm using an async map so the requests can be sent in parallel instead of sequentially. Still quite ugly in my opinion. I might have to rework my database structure (or ditch firebase completely) to improve this.